### PR TITLE
Fixes #169 content type

### LINKED ISSUE #169: ContentType has no effect in 2.0-preview
### Description

The attribute contentType is not taken into account on post.

Example:

      <iron-form id="myForm">
        <form method="post" action="/a/valid/url" content-type="application/json">
          <paper-input id="foo" label="Foo" name="foo" value="fooValue"></paper-input>
          <paper-input id="bar" label="bar" name="bar" value="barValue"></paper-input>

          <paper-button id="mybutton" raised>Action</paper-button>
          <input type="submit" hidden>
        </form>
      </iron-form>

### Expected outcome

The code above is expected to post an Json '{"foo":"fooValue","bar":"barValue"}' with the reques-headers content-type set to "application/json". 

### Actual outcome

But we got `Content-Type:text/plain;charset=UTF-8`

### Browsers Affected

- [x] Chrome
- [ ] Firefox
- [ ] Safari 9
- [ ] Safari 8
- [ ] Safari 7
- [ ] Edge
- [ ] IE 11
- [ ] IE 10

### PR DESCRIPTION
**Pull request by hubjac1**
This fixes #169 by reading content-type from from attributes.

### CODE REVIEW

**Comment by hubjac1** [comment]
Travis CI build was already failed for this branch. 
Do I need to fix before merge?

**Comment by notwaldorf** [comment]
@hubjac1 You need to sign the CLA before I can merge this PR

**Comment by notwaldorf** [comment]
Oh nevermind, CLA looks good:
screen shot 2017-05-24 at 12 02 29

File: iron-form.html [hunk 1]
```diff
@@ -274,9 +274,9 @@
         // elements with those names.
         this.request.url = this._form.getAttribute('action');
         this.request.method = this._form.getAttribute('method');
-        this.request.contentType = this._form.contentType;
-        //this.request.withCredentials = this.withCredentials;
-        //this.request.headers = this.headers;
+        this.request.contentType = this._form.getAttribute('content-type');
```
Comment (notwaldorf) [inline iron-form.html]: i think this may be `enctype` actually

**Comment by notwaldorf** [comment]
Oh hmm now after  fixed the conflicts i kept `enctype` in the iron-form code, but haven't updated the tests -- would you mind doing that?

**Comment by hubjac1** [comment]
@notwaldorf  Thanks, If you can wait until Monday, i will be please to help.

**Comment by notwaldorf** [comment]
@hubjac1 absolutely!

**Comment by hubjac1** [comment]
@notwaldorf here you are.

File: test/basic.html [hunk 2]
```diff
@@ -721,6 +740,73 @@
...
+        setup(function() {
+            server = sinon.fakeServer.create();
+            f = fixture('content-type');
+        });
+
+        teardown(function() {
+            server.restore();
+        });
+
+        test('submits a form', function(done) {
```
Comment (notwaldorf) [inline test/basic.html]: This indentation seems whack, but I'll merge this PR and fix it myself, so that I don't keep you waiting :)

**Pull request merged**